src/modules/ratings/rating.service.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { sumBy } from 'lodash';
+import { Model } from 'mongoose';
+import { GetQueryRatingDto } from './dto/get-query-rating.dto';
+import { RatingDto } from './dto/rating.dto';
+import { Rating } from './schemas/rating.schema';
+
+@Injectable()
+export class RatingService {
+  constructor(
+    @InjectModel(Rating.name)
+    private ratingModel: Model<Rating>
+  ) {}
+
+  createRating(newRating: RatingDto): Promise<Rating> {
+    return this.ratingModel.create(newRating);
+  }
+
+  getRatings(filter: GetQueryRatingDto): Promise<Rating[]> {
+    return this.ratingModel.find({
+      ...(filter.workoutId ? { workoutId: filter.workoutId } : {}),
+      ...(filter.athleteId ? { athleteId: filter.athleteId } : {}),
+      ...(filter.rating ? { rating: filter.rating } : {}),
+    });
```

Comment:
si le pasas filter solo deberia funcionar

Reply:
no funciona eso

Reply:
que probaste? Si haces `return this.ratingModel.find(filter || {})` no funciona?